models/aeif_psc_delta.h

```diff
@@ -0,0 +1,391 @@
+/*
+ *  aeif_psc_delta.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef AEIF_PSC_delta_H
+#define AEIF_PSC_delta_H
+
+// Generated includes:
+#include "config.h"
+
+#ifdef HAVE_GSL
+
+// External includes:
+#include <gsl/gsl_errno.h>
+#include <gsl/gsl_matrix.h>
+#include <gsl/gsl_odeiv.h>
+
+// Includes from nestkernel:
+#include "archiving_node.h"
+#include "connection.h"
+#include "event.h"
+#include "nest_types.h"
+#include "recordables_map.h"
+#include "ring_buffer.h"
+#include "universal_data_logger.h"
+
+/* BeginDocumentation
+Name: aeif_psc_delta - Current-based adaptive exponential integrate-and-fire
+neuron
+                      model according to Brette and Gerstner (2005) with delta
+synapse.
+
+Description:
+
+aeif_psc_delta is the adaptive exponential integrate and fire neuron
+according to Brette and Gerstner (2005), with post-synaptic currents
+in the form of delta spikes.
+
+This implementation uses the embedded 4th order Runge-Kutta-Fehlberg
+solver with adaptive stepsize to integrate the differential equation.
+
+The membrane potential is given by the following differential equation:
+C dV/dt= -g_L(V-E_L)+g_L*Delta_T*exp((V-V_T)/Delta_T)+I_ex(t)+I_in(t)+I_e
+
+and
+
+tau_w * dw/dt= a(V-E_L) -W
+
+I_x(t) = J_x Sum_k delta(t - t^x_k),
+
+where x = (ex, in), delta is the dirac delta function and k indexes incoming
```

Comment:
For the psc_delta variant, it makes no sense to differentiate between ex and in, since there are no time constants. There should just be a single `I_syn`; code is fine, but this comment needs to be cleaned up.